### PR TITLE
Bug 2067740: update prometheus/client_golang version

### PR DESCRIPTION
**Pull request by raptorsun**
This PR upgrades prometheus/client-golang to version 1.12.1, thus resolving bug [2067740](https://bugzilla.redhat.com/show_bug.cgi?id=2067740).

* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,7 +14,7 @@ require (
 	github.com/prometheus-operator/prometheus-operator v0.53.1
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.53.1
 	github.com/prometheus-operator/prometheus-operator/pkg/client v0.53.1
-	github.com/prometheus/client_golang v1.11.0
+	github.com/prometheus/client_golang v1.11.1
```
Comment (simonpasquier) [inline go.mod:17]: we can bump to the latest stable release.
```suggestion
	github.com/prometheus/client_golang v1.12.1
```

Comment (raptorsun) [reply go.mod:17]: Can we assume that every tag in [the Tag tab](https://github.com/prometheus/client_golang/tags) is stable release?

Comment (raptorsun) [reply go.mod:17]: I find it ,the release list is not in an increasing order 😅 never mind

**Comment by simonpasquier** [comment]
/retitle Bug 2067740: update prometheus/client_golang version

**Comment by raptorsun** [comment]
The package prometheus/client-golang has been upgraded to version 1.12.1.

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Pull request merged**